fabtests/functional/rdm_tagged_peek.c

```diff
@@ -278,15 +278,24 @@ static int do_recvs(void)
 static int do_sends(void)
 {
 	int i, ret;
+	struct fi_msg_tagged tmsg;
+	struct iovec iov;
+
+	iov.iov_base = tx_buf;
+	iov.iov_len = tx_size;
+	tmsg.msg_iov = &iov;
+	tmsg.iov_count = 1;
+	tmsg.desc = &mr_desc;
+	tmsg.addr = remote_fi_addr;
 
 	printf("Sending %d tagged messages\n", SEND_CNT);
 	(void) ft_fill_buf(tx_buf, tx_size);
 
 	for (i = 1; i <= SEND_CNT; i++) {
 		do {
-			ret = fi_tsend(ep, tx_buf, tx_size, mr_desc,
-					remote_fi_addr, BASE_TAG + i,
-					&tx_ctx_arr[i].context);
+			tmsg.tag = BASE_TAG + i;
+			tmsg.context = &tx_ctx_arr[i].context;
+			ret = fi_tsendmsg(ep, &tmsg, FI_TRANSMIT_COMPLETE);
```

Comment:
Changing a test that doesn't appear to be broken, so that a provider can be made to pass the test kind of defeats part of the purpose for having the tests...

This flag results in the ucx provider calling ucp_worker_flush().  That's also called when closing the ep.

Reply:
The test causes a segfault.  That doesn't sound like a false positive, even if the crash isn't related to the PR.  The test can be disabled for ucx, if needed.  If the problem is outside of the provider, then disabling the test is reasonable.

Reply:
Yes, I agree that the test should be disabled if it's outside of the provider. I'm also suggesting a modification of the provider (see above) that might allow us to keep testing this for the provider. Do you or @j-xiong have thoughts about the above variable?

Reply:
I don't know what ucp_worker_flush() does, but always calling it from ep close sounds reasonable.  (Remove the if statement).  The changes to the test result in that function being called, which suggests that there's something relative to ucp_worker_flush() being called that's needed.